app/helpers/react_components/mentoring/solution.rb

```diff
@@ -0,0 +1,174 @@
+              }
+            },
+            track: {
+              title: track.title,
+              highlightjs_language: track.highlightjs_language,
+              icon_url: track.icon_url
+            },
+            exercise: {
+              title: exercise.title
+            },
+            iterations: iterations,
+            mentor_solution: mentor_solution,
+            notes: notes,
+            links: links,
+            relationship: SerializeMentorStudentRelationship.(student_mentor_relationship),
+            request: request_json,
+            discussion: discussion_json
+          }
+        )
+      end
+
+      memoize
+      def request
+        ::Solution::MentorRequest.find_by(solution: solution)
+      end
+
+      def request_json
+        return if request.blank?
+
+        {
+          id: request.uuid,
+          comment: request.comment,
+          updated_at: request.updated_at.iso8601,
+          is_locked: request.locked?,
+          links: {
+            lock: Exercism::Routes.lock_api_mentor_request_path(request),
+            discussion: Exercism::Routes.api_mentor_discussions_path
+          }
+        }
+      end
+
+      def discussion
+        ::Solution::MentorDiscussion.find_by(solution: solution, mentor: current_user)
+      end
+
+      def student_mentor_relationship
+        Mentor::StudentRelationship.find_by(mentor: current_user, student: student)
+      end
+
+      def discussion_json
+        return if discussion.blank?
+
+        {
+          id: discussion.uuid,
+          is_finished: discussion.finished?,
+          links: { posts: Exercism::Routes.api_mentor_discussion_posts_url(discussion) }.tap do |links|
+            if discussion.requires_mentor_action?
+              links[:mark_as_nothing_to_do] =
+                Exercism::Routes.mark_as_nothing_to_do_api_mentor_discussion_path(discussion)
+            end
+
+            links[:finish] = Exercism::Routes.finish_api_mentor_discussion_path(discussion) unless discussion.finished?
+          end
+        }
+      end
+
+      def links
+        {
+          mentor_dashboard: Exercism::Routes.mentor_dashboard_path,
+          exercise: Exercism::Routes.track_exercise_path(track, exercise),
+          scratchpad: Exercism::Routes.api_scratchpad_page_path(scratchpad.category, scratchpad.title)
+        }
+      end
+
+      def iterations
+        comment_counts = if discussion
+                           ::Solution::MentorDiscussionPost.where(discussion: discussion).
+                             group(:iteration_id, :seen_by_mentor).count
+                         end
+
+        solution.iterations.map do |iteration|
+          discussion ? counts = comment_counts.select { |(it_id, _), _| it_id == iteration.id } : counts = nil
+          discussion ? num_comments = counts.sum(&:second) : num_comments = 0
+          discussion ? unread = counts.reject { |(_, seen), _| seen }.present? : unread = 0
```

Comment:
Is there a reason not do:

```suggestion
          counts = discussion ? comment_counts.select { |(it_id, _), _| it_id == iteration.id } : nil
          num_comments = discussion ? counts.sum(&:second) : 0
          unread = discussion ? counts.reject { |(_, seen), _| seen }.present? : 0
```

This code does not have the duplicate assignments, which I found slightly hard to read. But if the original code is idiomatic Ruby, just ignore :)

Reply:
I suspect this is Rubocop. I find it's rules about this really weird, so might just disable them.

Reply:
> I suspect this is Rubocop. I find it's rules about this really weird, so might just disable them.

Yes, that is Rubocop being silly.  It doesn't make sense to do unconditional things in conditional branches. 🤮 

```ruby
         counts = comment_counts.select { |(it_id, _), _| it_id == iteration.id } if discussion
         num_comments = discussion ? counts.sum(&:second) : 0
         unread = discussion ? counts.reject { |(_, seen), _| seen }.present? : 0
```

The first line having a `nil` for _false_ branch should probably be a guard statement.  It will be nil if discussion is false or nil.

The Rubocop for this:

```yml
Style/ConditionalAssignment:
  Enabled: true
  EnforcedStyle: assign_to_condition
  SingleLineConditionsOnly: false
  IncludeTernaryExpressions: true
```

Reply:
Great. Thanks, @kotp :)